File: backend/src/constants/hazard_severity.py
HazardSeverity = ["LOW", "MEDIUM", "HIGH", "CRITICAL"]

# 严重程度决定责任人和整改期限：级别越高，责任人层级越高、期限越短。
SEVERITY_POLICY = {
  "LOW": {"owner_id": 101, "deadline_days": 30},
  "MEDIUM": {"owner_id": 201, "deadline_days": 7},
  "HIGH": {"owner_id": 301, "deadline_days": 3},
  "CRITICAL": {"owner_id": 401, "deadline_days": 1}
}

# 合并隐患单时用于比较严重度高低。
SEVERITY_ORDER = {"LOW": 0, "MEDIUM": 1, "HIGH": 2, "CRITICAL": 3}

File: frontend/src/constants/HazardSeverity.ts
export const HazardSeverity = ["LOW","MEDIUM","HIGH","CRITICAL"] as const;
export type HazardSeverity = (typeof HazardSeverity)[number];
export const HazardSeverityText: Record<HazardSeverity, string> = Object.fromEntries(HazardSeverity.map((value) => [value, value.replace(/_/g, " ")])) as Record<HazardSeverity, string>;

// 与后端 constants/hazard_severity.py 的 SEVERITY_POLICY 保持一致：严重程度决定责任人和期限。
export const HazardSeverityPolicy: Record<HazardSeverity, { ownerId: number; ownerLabel: string; deadlineDays: number }> = {
  LOW: { ownerId: 101, ownerLabel: "维保员", deadlineDays: 30 },
  MEDIUM: { ownerId: 201, ownerLabel: "维保班长", deadlineDays: 7 },
  HIGH: { ownerId: 301, ownerLabel: "维保主管", deadlineDays: 3 },
  CRITICAL: { ownerId: 401, ownerLabel: "物业工程负责人", deadlineDays: 1 }
};
